integrations/pytorch/train.py

```diff
@@ -0,0 +1,406 @@
+                        schedule to apply them with. Can also provide a
+                        SparseZoo stub prefixed with 'zoo:' with an optional
+                        '?recipe_type=' argument
+  --sparse-transfer-learn
+                        Enable sparse transfer learning modifiers to enforce
+                        the sparsity for already sparse layers. The modifiers
+                        are added to the ones to be loaded from the recipe-
+                        path
+  --eval-mode           Puts into evaluation mode so that the model can be
+                        evaluated on the desired dataset
+  --train-batch-size TRAIN_BATCH_SIZE
+                        The batch size to use while training
+  --test-batch-size TEST_BATCH_SIZE
+                        The batch size to use while testing
+  --optim OPTIM         The optimizer type to use, one of [SGD, Adam]
+  --logs-dir LOGS_DIR   The path to the directory for saving logs
+  --save-best-after SAVE_BEST_AFTER
+                        start saving the best validation result after the
+                        given epoch completes until the end of training
+  --save-epochs SAVE_EPOCHS [SAVE_EPOCHS ...]
+                        epochs to save checkpoints at
+  --use-mixed-precision
+                        Trains model using mixed precision. Supported
+                        environments are single GPU and multiple GPUs using
+                        DistributedDataParallel with one GPU per process
+  --debug-steps DEBUG_STEPS
+                        Amount of steps to run for training and testing for a
+                        debug mode
+#########
+EXAMPLES
+#########
+
+##########
+Example command for pruning resnet50 on imagenet dataset:
+python integrations/pytorch/train.py \
+    --recipe-path ~/sparseml_recipes/pruning_resnet50.yaml \
+    --arch-key resnet50 --dataset imagenet --dataset-path ~/datasets/ILSVRC2012 \
+    --train-batch-size 256 --test-batch-size 1024
+
+##########
+Example command for transfer learning sparse mobilenet_v1 on an image folder dataset:
+python integrations/pytorch/train.py \
+    --sparse-transfer-learn \
+    --recipe-path  ~/sparseml_recipes/pruning_mobilenet.yaml \
+    --arch-key mobilenet_v1 --pretrained pruned-moderate \
+    --dataset imagefolder --dataset-path ~/datasets/my_imagefolder_dataset \
+    --train-batch-size 256 --test-batch-size 1024
+
+##########
+Template command for running training with this script on multiple GPUs using
+DistributedDataParallel using mixed precision. Note - DDP support in this script
+only tested for torch==1.7.0.
+python -m torch.distributed.launch \
+--nproc_per_node <NUM GPUs> \
+integrations/pytorch/train.py \
+--use-mixed-precision \
+<VISION.PY TRAIN ARGUMENTS>
+"""
+import argparse
+from typing import Any, List, Tuple
+
+import torch
+from torch.nn import Module
+from torch.utils.data import DataLoader
+
+from helpers import (
```

Comment:
I am not 100% familiar with the import structure in `sparseml` but would this import work if one tried to execute the script from another directory? 
Or is that not supported?

Reply:
It should work when executed from a different directory @lucaslie 
```python
import sys
print(sys.path)
```

The train, export, analysis scripts have '/home/rahul/projects/sparseml/integrations/pytorch' in their `sys.path` so python should be able to find `helpers.py`, it would work as long as helpers is not in a separate directory. 

Try running the following command

```bash
python sparseml/integrations/pytorch/analysis.py lr_sensitivity \
    --arch-key mobilenet --dataset imagenette \
    --dataset-path ~/datasets --batch-size 2
```

It takes about ~20secs to run. It works on my machine, let me know if you get an error, will change this to a relative import then if needed.

Reply:
Sounds good. I will check it before submitting my review.

Reply:
yes, that works :)